### PR TITLE
dns: Remove parser buffering code

### PR DESCRIPTION
**Pull request by jlucovsky**
This PR removes the buffering code from the DNS TCP parsers.

`make check` Rust tests succeed; `suricata-verify` tests also succeed.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3538](https://redmine.openinfosecfoundation.org/issues/3538)

Describe changes:
- Move DNS query splice test to Rust.

### CODE REVIEW

File: src/detect-dns-query.c [hunk 1]
```diff
@@ -1256,13 +929,9 @@ static void DetectDnsQueryRegisterTests(void)
     UtRegisterTest("DetectDnsQueryTest01", DetectDnsQueryTest01);
     UtRegisterTest("DetectDnsQueryTest02", DetectDnsQueryTest02);
     UtRegisterTest("DetectDnsQueryTest03 -- tcp", DetectDnsQueryTest03);
-    UtRegisterTest("DetectDnsQueryTest04 -- tcp splicing",
-                   DetectDnsQueryTest04);
```
Comment (catenacyber) [inline src/detect-dns-query.c:1260]: Why remove these tests ?

Comment (jlucovsky) [reply src/detect-dns-query.c:1260]: A replacement for `DetectBsizeSigTest01` was added to the Rust test set. These tests require TCP buffering which isn't possible when directly invoking the application layer parsing.

Comment (catenacyber) [reply src/detect-dns-query.c:1260]: > These tests require TCP buffering which isn't possible when directly invoking the application layer parsing.

It is possible as I did recently when translating the SSH parser to Rust and using ApplayerResult::Incomplete
You can see in `DetectSshVersionTestDetect01` cf https://github.com/OISF/suricata/blob/master/src/detect-ssh-proto-version.c#L339

And I think it is particularly useful to keep these tests when doing such a change ;-)

Comment (jasonish) [reply src/detect-dns-query.c:1260]: From the DNS perspective I think its probably OK to not test this as a unit test. We should have tests that that check the correct response is returned when the data is incomplete.

While there should be some larger tests that test that the buffering +  protocol parsing does work, it probably does not need to be implemented for every protocol.

What might be more intersting is suricata-verify tests with pcaps that cause the parsers to be called with incomplete data.

Comment (victorjulien) [reply src/detect-dns-query.c:1260]: Related: https://redmine.openinfosecfoundation.org/issues/3725

Comment (catenacyber) [reply src/detect-dns-query.c:1260]: I agree that they can be replaced by suricata-verify tests.

I wish I had known this ticket while translating the SSH tests ;-)

**Comment by victorjulien** [comment]
Can you rebase and resubmit?

**Comment by jlucovsky** [comment]
Continued in #5020

**Pull request closed without merging**